### PR TITLE
Memoize MakeGenericMethod(), MakeGenericType() methods (#50)

### PR DESCRIPTION
**Pull request by SergeiPavlov**

### CODE REVIEW

**Review by alex-kulakov — CHANGES_REQUESTED** [review]

File: Orm/Xtensive.Orm/Orm/Building/Builders/ModelBuilder.cs [hunk 1]
```diff
@@ -403,20 +424,21 @@ private void BuildAuxiliaryTypes(IEnumerable<AssociationInfo> associations)
...
-      var result = GeneratedTypes.GetValue(typeName,
-        (_typeName, _baseType) =>
-          TypeHelper.CreateInheritedDummyType(_typeName, _baseType, true),
-        baseType);
+      static Lazy<Type> AuxiliaryTypeFactory(TypeKey typeKey) {
+        return new Lazy<Type>(() => {
+          var baseType = WellKnownOrmTypes.EntitySetItemOfT1T2.CachedMakeGenericType(typeKey.OwnerType, typeKey.TargetType);
+          return TypeHelper.CreateInheritedDummyType(typeKey.Name, baseType, true);
+        });
+      };
```
[the remark this answers is not in the document]
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Building/Builders/ModelBuilder.cs]: Moved it to static readonly as most optimal way

File: Orm/Xtensive.Orm/Sql/SqlDml.cs [hunk 2]
```diff
@@ -19,6 +20,10 @@ namespace Xtensive.Sql
   /// </summary>
   public static class SqlDml
   {
+    private static readonly Type
+      sqlArrayType = typeof(SqlArray<>),
+      sqlLiteralType = typeof(SqlLiteral<>);
```
[the remark this answers is not in the document]
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Sql/SqlDml.cs]: done
Actually it is called `PascalCase`
`sqlArrayType` was in `camelCase`

**Review by alex-kulakov — DISMISSED** [review]

File: Orm/Xtensive.Orm/Orm/Building/Builders/AutoGenericCombinator.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
[the remark this answers is not in the document]
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Building/Builders/AutoGenericCombinator.cs:1]: I'm sure it  was done automatically by Editor.

[the remark this answers is not in the document]
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Building/Builders/AutoGenericCombinator.cs]: reverted it

File: Orm/Xtensive.Orm/Sql/SqlDml.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-kulakov) [reply Orm/Xtensive.Orm/Sql/SqlDml.cs]: Yep, I mismatch them, but you get it right, thank you

**Review by alex-kulakov — APPROVED** [review]

**Pull request merged**